homeassistant/components/overkiz/__init__.py

```diff
@@ -139,8 +191,14 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
-
     if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        # Remove local auth token if local server is unloaded
+        if token_uuid := entry.data.get(CONF_TOKEN_UUID):
+            data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
+            await data.coordinator.client.delete_local_token(
+                cast(str, entry.unique_id), token_uuid
+            )
+
         hass.data[DOMAIN].pop(entry.entry_id)
```

Comment:
This will still throw the following error about "Session is closed". Is the issue that the delete_local_token is async and the `pop()` happens too soon? Or am I overlooking something?

```
2023-06-27 19:27:37.815 ERROR (MainThread) [homeassistant.config_entries] Error unloading entry gateway-1225-7298-3293.local:8443 for overkiz
Traceback (most recent call last):
  File "/workspaces/home-assistant-core/homeassistant/config_entries.py", line 546, in async_unload
    result = await component.async_unload_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/home-assistant-core/homeassistant/components/overkiz/__init__.py", line 198, in async_unload_entry
    data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
  File "/home/vscode/.local/lib/python3.11/site-packages/backoff/_async.py", line 151, in retry
    ret = await target(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/pyoverkiz/client.py", line 738, in delete_local_token
    await self.__delete(f"config/{gateway_id}/local/tokens/{uuid}")
  File "/home/vscode/.local/lib/python3.11/site-packages/pyoverkiz/client.py", line 801, in __delete
    async with self.session.delete(
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
                 ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 400, in _request
    raise RuntimeError("Session is closed")
RuntimeError: Session is closed
```